### ISSUE TITLE
superfluous response.WriteHeader call

### ISSUE THREAD

**Issue by nvtkaszpir** [prometheus/graphite_exporter#123]
We got graphite_exporter as a sidecar container in k8s pod, the app sends metrics in graphite format to the graphite_exporter sidecar. Unfortunately the sidecar is sometimes crashing with messages as below (notice the start messages, because pod was restarted):

```text
E 2020-03-26T08:30:46.433181765Z 2020/03/26 08:30:46 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:30:47.290749036Z 2020/03/26 08:30:47 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:30:47.471443864Z 2020/03/26 08:30:47 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:30:48.251203872Z 2020/03/26 08:30:48 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:30:56.142616995Z 2020/03/26 08:30:56 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:31:06.131707428Z 2020/03/26 08:31:06 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:31:16.130500694Z 2020/03/26 08:31:16 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:31:26.137004953Z 2020/03/26 08:31:26 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:31:36.128960391Z 2020/03/26 08:31:36 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:31:46.216874310Z 2020/03/26 08:31:46 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:31:56.170481822Z 2020/03/26 08:31:56 http: superfluous response.WriteHeader call from github.com/prometheus/client_golang/prometheus/promhttp.(*responseWriterDelegator).WriteHeader (delegator.go:59)
E 2020-03-26T08:31:59.369481547Z time="2020-03-26T08:31:59Z" level=info msg="Starting graphite_exporter (version=0.6.0, branch=HEAD, revision=66c1c5e4eea14cf7f22080afcb9740ce4d874ffc)" source="main.go:245"
E 2020-03-26T08:31:59.369552488Z time="2020-03-26T08:31:59Z" level=info msg="Build context (go=go1.12.5, user=root@5cd57e93a5e2, date=20190531-17:44:40)" source="main.go:246"
E 2020-03-26T08:31:59.369562624Z time="2020-03-26T08:31:59Z" level=info msg="Listening on :9108" source="main.go:321"
```

related issues:
- https://github.com/prometheus/prometheus/issues/6139
- https://github.com/prometheus/statsd_exporter/issues/257

Fix:
- probably bump prometheus/client_golang to version 1.5.1  in the graphite_exporter, adjust code and so on...

**Comment by matthiasr** [comment]
yes that's right, thank you!

**Comment by matthiasr** [comment]
fixed by #125
